### ISSUE TITLE
Unable to initialize IAM sub-system, some users may not be available: signature is invalid

### ISSUE THREAD

**Issue by clcc2019** [minio/minio#16934]
## NOTE
If this case is urgent, please subscribe to [Subnet](https://min.io/pricing) so that our 24/7 support team may help you faster.

## Expected Behavior
minio upgrade to latest 

The IAM sub-system should initialize without errors allowing all users to be available for use.

## Current Behavior
The following error message occurs: `Unable to initialize IAM sub-system, some users may not be available: signature is invalid`. This indicates that there is an issue with the AWS credentials being used to access the IAM (Identity and Access Management) sub-system. The error message specifically mentions an invalid signature, which typically means that the access key or secret key being used is incorrect or has expired.

## Possible Solution
To resolve the issue, please try the following:
- Double-check the access key and secret key to make sure they are correct and still valid.
- Ensure that the IAM user associated with the access key and secret key has the necessary permissions to perform the desired action.
- Confirm that the system clock on your local machine or server is synced with NTP servers. An incorrect time can cause authentication-related errors.

## Steps to Reproduce (for bugs)
N/A - This error occurs when initializing the IAM sub-system.

## Context
This error may prevent certain users from being available for use and could potentially limit functionality within the application.

## Regression
N/A

## Your Environment
- Version used (`minio --version`): latest
- Server setup and configuration:
- Operating System and version (`uname -a`): Linux 13f3b44c70bf 3.10.0-1160.53.1.el7.x86_64 #1 SMP Fri Jan 14 13:59:45 UTC 2022 x86_64 GNU/Linux

**Comment by harshavardhana** [comment]
@clcc2019 you are not explaining what you did here, did you change root credentials?

**Comment by clcc2019** [comment]
> @clcc2019 you are not explaining what you did here, did you change root credentials?

I did not modify root
<img width="1506" alt="image" src="https://user-images.githubusercontent.com/46099295/229013122-b19b4411-fb3e-4aa9-a88a-d22e54126e40.png">

**Comment by harshavardhana** [comment]
I am pretty sure you changed the root credentials when starting MinIO - https://github.com/minio/minio/pull/16812 should fix it.

You should upgrade.

**Comment by olevitt** [comment]
Hi ! We are facing the exact same issue since we upgraded from `RELEASE.2023-01-31T02-24-19Z` to `RELEASE.2023-04-13T03-08-07Z` (we had to upgrade due to being vulnerable to `Cve-2023-28432`).  
Now, each time there is a cluster restart (e.g after any update) the IAM subsystem gets corrupted with error `Unable to initialize IAM sub-system, some users may not be available: signature is invalid` and nothing related to IAM / STS works anymore. Wiping `.minio.sys` folder gets it running again (with side-effect of losing every service account and policy) ... until next restart.  
We tried upgrading to the latest version `RELEASE.2023-05-04T21-44-30Z` and it corrupted the IAM subsystem once again ...
So this bug is definitely not fixed
